nutils/numeric.py

```diff
@@ -676,4 +676,20 @@ def sanitize_einsum_subscripts(subscripts, *shapes):
     return (*in_, out)
 
 
+def compress_indices(indices, length):
```

Comment:
Suggestion: It took me a bit of time to understand that the returned array is an array of insertion positions of `range(length+1)` in `indices`. What about

```python
'''Returns the insertion positions of ``range(length+1)`` in the monotonically increasing ``indices``.'''
```

Reply:
Changed it to *Return insertion positions in monotonically increasing index vector* for brevity, with the following elaboration:

The compressed array ``c`` is such that ``indices[c[i]:c[i+1]] == i`` for any integer ``0 <= i <= length``. Fails with ``ValueError`` if ``indices`` are out of bounds or not monotonically increasing; otherwise fully equivalent to ``indices.searchsorted(numpy.arange(length + 1))``.